### PR TITLE
Gdpr mam muc schema riak

### PR DESCRIPTION
**Pull request by NelsonVides**
This PR addresses riak schema change for gdpr mam muc retrieval

### CODE REVIEW

File: src/mam/mod_mam_riak_timed_arch_yz.erl [hunk 1]
```diff
@@ -337,7 +337,7 @@ get_mam_pm_gdpr_data(Username, Host) ->
     Jid = jid:make({LUser, LServer, <<>>}),
     LookupParams = ?DUMMY_LOOKUP_PARAMETERS#{owner_jid := Jid},
     {ok, {_, _, Messages}} = lookup_messages([], Host, LookupParams),
-    {ok, [{Id, jid:to_binary(Jid), exml:to_binary(Packet)} || {Id, Jid, Packet} <- Messages]}.
+    {ok, [{Id, jid:to_binary(Jid2), exml:to_binary(Packet)} || {Id, Jid2, Packet} <- Messages]}.
```
Comment (NelsonVides) [inline src/mam/mod_mam_riak_timed_arch_yz.erl]: This is just taking an opportunity to improve a lint, that `Jid` was shadowing a previous variable.

**Comment by mongoose-im** [comment]
[6655.1](https://travis-ci.org/esl/MongooseIM/jobs/541616610) / Erlang 19.3 / small_tests / aa02d427b0fadfd6165993a866c3d3282e53e40a
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2330/6655/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2330/6655/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-06-05_08.28.10/lib.mongooseim.logs/run.2019-06-05_08.28.11/suite.log.html)

File: src/mam/mod_mam_riak_timed_arch_yz.erl [hunk 2]
```diff
@@ -347,11 +347,14 @@ get_mam_muc_gdpr_data(Username, Host) ->
     Jid = jid:make({LUser, LServer, <<>>}),
     LookupParams = ?DUMMY_LOOKUP_PARAMETERS#{with_jid := Jid},
     {ok, {_, _, Messages}} = lookup_messages([], Host, LookupParams),
-    Filtered = lists:filter(fun(El) -> is_muclight_message(Jid, El) end, Messages),
+    Filtered = lists:filter(fun({_,_,Msg}) -> is_groupchat_message(Msg) end, Messages),
```
Comment (NelsonVides) [inline src/mam/mod_mam_riak_timed_arch_yz.erl]: Previously, `lookup_messages` is extracting all messages that can be filtered with `with_jid := Jid` in the parameters. But these messages can be muc messages, and also private messages. We were previously matching on the nickname (that is what muclight is extracting, if the resource matches the jid, it is a muclight message, but then it wouldn't match classic mucs. What they both have in common, against private message, is the type, not the jids.

**Comment by mongoose-im** [comment]
[6656.1](https://travis-ci.org/esl/MongooseIM/jobs/541647658) / Erlang 19.3 / small_tests / 24d76f642c19d04dd218775d519c4461037309c2
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2330/6656/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2330/6656/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-06-05_09.49.32/lib.mongooseim.logs/run.2019-06-05_09.49.32/suite.log.html)

File: src/mam/mod_mam_riak_timed_arch_yz.erl [hunk 3]
```diff
@@ -132,10 +132,10 @@ archive_message(_Result, Host, MessId, _UserID, LocJID, RemJID, SrcJID, _Dir, Pa
             {error, Reason}
     end.
 
-archive_message_muc(_Result, Host, MessId, _UserID, LocJID, _FromJID, SrcJID, _Dir, Packet) ->
-    RemJIDMuc = maybe_muc_jid(SrcJID),
+archive_message_muc(_Result, Host, MessId, _UserID, LocJID, FromJID, SrcJID, _Dir, Packet) ->
+    BareFromJID = jid:to_bare(FromJID),
     try
-        archive_message(Host, MessId, LocJID, RemJIDMuc, SrcJID, Packet, muc)
+        archive_message(Host, MessId, LocJID, BareFromJID, SrcJID, Packet, muc)
```
Comment (NelsonVides) [inline src/mam/mod_mam_riak_timed_arch_yz.erl]: This is the real schema change:
`LocJID` is the room's JID, i.e., 'room@domain'
`FromJID` is the real user's JID, i.e., 'user@domain/resource'
`SrcJID` is the user's JID in the room, i.e., 'room@domain/nickname'
If we do `maybe_muc_jid(SrcJID)`, of something with a resource, we will archive by this resource. But we want to archive by the user, so we can query so later. Unfortunately this makes it archive by the same pattern than private messages, so we need to filter later by type, as we do in this PR.

File: src/mam/mod_mam_riak_timed_arch_yz.erl [hunk 4]
```diff
@@ -347,11 +347,11 @@ get_mam_muc_gdpr_data(Username, Host) ->
...
     LookupParams = ?DUMMY_LOOKUP_PARAMETERS#{with_jid := Jid},
     {ok, {_, _, Messages}} = lookup_messages([], Host, LookupParams),
-    Filtered = lists:filter(fun(El) -> is_muclight_message(Jid, El) end, Messages),
+    Filtered = lists:filter(fun({_,_,Msg}) -> is_groupchat_message(Msg) end, Messages),
     {ok, [{MsgId, exml:to_binary(Packet)} || {MsgId, _, Packet} <- Filtered]}.
 
-is_muclight_message(_BareJid, {_MsgId, #jid{lresource = <<>>}, _Packet})    -> false;
-is_muclight_message(BareJid, {_MsgId, #jid{lresource = Resource}, _Packet}) -> jid:to_binary(BareJid) == Resource.
+is_groupchat_message(Msg) ->
+    exml_query:attr(Msg, <<"type">>, undefined) == <<"groupchat">>.
```
Comment (NelsonVides) [inline src/mam/mod_mam_riak_timed_arch_yz.erl]: `==` or `=:=`??

**Comment by mongoose-im** [comment]
[6657.1](https://travis-ci.org/esl/MongooseIM/jobs/541698777) / Erlang 19.3 / small_tests / acf04b9635d8c0cb7e88d78b742db4c45a275499
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2330/6657/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2330/6657/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-06-05_11.57.54/lib.mongooseim.logs/run.2019-06-05_11.57.55/suite.log.html)

**Comment by mongoose-im** [comment]
[6661.1](https://travis-ci.org/esl/MongooseIM/jobs/542043174) / Erlang 19.3 / small_tests / 4cbfceae2f636efe6338a895b67026a9f7f78a19
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2330/6661/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2330/6661/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-06-06_02.25.30/lib.mongooseim.logs/run.2019-06-06_02.25.31/suite.log.html)

**Comment by DenysGonchar** [comment]
looking at your changes, i noticed that they are breaking MAM MUC queries with [Filtering by JID](https://xmpp.org/extensions/xep-0313.html#filter-jid)

I was surprised that it's not cached by MAM SUITE, but it happened so because `muc_querying_for_all_messages_with_jid` test case is broken and returns false positive result.

I fixed the test case in this [commit](https://github.com/esl/MongooseIM/pull/2330/commits/8a53fd3dc89f9fcd9611a922ff4e91c0bdf6ca05) now it's your turn to fix the changes in DB schema

**Comment by mongoose-im** [comment]
[6665.1](https://travis-ci.org/esl/MongooseIM/jobs/542817450) / Erlang 19.3 / small_tests / e931facefc602dabfd1de459741d540d96b3ee40
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2330/6665/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2330/6665/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-06-07_15.19.15/lib.mongooseim.logs/run.2019-06-07_15.19.15/suite.log.html)

**Comment by mongoose-im** [comment]
[6668.1](https://travis-ci.org/esl/MongooseIM/jobs/543610727) / Erlang 19.3 / small_tests / d11d72a464a7d871575d10de727e63c46224f332
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2330/6668/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2330/6668/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-06-10_07.25.33/lib.mongooseim.logs/run.2019-06-10_07.25.33/suite.log.html)

**Comment by mongoose-im** [comment]
[6675.1](https://travis-ci.org/esl/MongooseIM/jobs/544085785) / Erlang 19.3 / small_tests / 1402345d96deb61b7392a9814b49d0f35bc53a35
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2330/6675/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2330/6675/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-06-11_08.33.24/lib.mongooseim.logs/run.2019-06-11_08.33.25/suite.log.html)

**Comment by mongoose-im** [comment]
[6676.1](https://travis-ci.org/esl/MongooseIM/jobs/544137823) / Erlang 19.3 / small_tests / 7f2e65165cd440aa281119f510a57c8628634492
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2330/6676/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2330/6676/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-06-11_10.06.27/lib.mongooseim.logs/run.2019-06-11_10.06.28/suite.log.html)

**Pull request closed without merging**